app/subvisual-blog/services/post_processor.rb

```diff
@@ -11,13 +11,25 @@ def initialize(post)
     def process
       extensions = { autolink: true, fenced_code_blocks: true, tables: true }
       redcarpet = Redcarpet::Markdown.new(SyntaxHighlightedRenderer, extensions)
-      post.processed_body = redcarpet.render(post.body)
+      post.processed_body = redcarpet.render(body_without_main_headings)
 
       return unless post.respond_to?(:processed_intro=)
 
       post.processed_intro = redcarpet.render(markdown_intro)
     end
 
+    def body_without_main_headings
+      if has_main_headings?
+        post.body.gsub(/^#/, "##")
```

Comment:
It's downsizing all of them, only if the post has primary ones `#`.
This decision was mostly to keep compatibility with existing posts (without having to manually go and edit them), and to prevent us from having to teach everyone about this change

A warning is a good idea, I'll try to figure out a way to do that

> which might not always be the correct thing to do

Is there any case you can think of?